Cython/Includes/numpy/__init__.pxd

```diff
@@ -331,6 +333,10 @@ cdef extern from "numpy/arrayobject.h":
     ctypedef unsigned long long npy_uint96
     ctypedef unsigned long long npy_uint128
 
+    ctypedef struct npy_half:
+        npy_uint16 value
```

Comment:
@robertwb Hi, I try to use structure to define the type as suggested. However the failure below is raised. 

```bash
~/W/g/cython ❯❯❯ CFLAGS=-O0 python runtests.py -vv --no-cleanup numpy_memoryview
Python 3.6.3 |Anaconda custom (64-bit)| (default, Oct  6 2017, 12:04:38)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]

Running tests against Cython 0.28.2 21d627c0bf499c56fa3ca2265687c6326daf4be6
Backends: c,cpp

runTest (__main__.CythonRunTestCase)
compiling (c) and running numpy_memoryview ...
=== C/C++ compiler output: ===
In file included from numpy_memoryview.c:594:
In file included from /Users/facai/Library/anaconda3/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/facai/Library/anaconda3/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
In file included from /Users/facai/Library/anaconda3/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1816:
/Users/facai/Library/anaconda3/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: "Using deprecated NumPy API, disable it by "          "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by " \
 ^
numpy_memoryview.c:3237:44: error: offsetof requires struct, union, or class type, '__pyx_t_5numpy_float16_t' (aka 'unsigned short') invalid
  {&__Pyx_TypeInfo_nn_npy_uint16, "value", offsetof(__pyx_t_5numpy_float16_t, value)},
                                           ^        ~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/lib/clang/9.0.0/include/stddef.h:120:24: note: expanded from macro 'offsetof'
#define offsetof(t, d) __builtin_offsetof(t, d)
                       ^                  ~
1 warning and 1 error generated.
```

I checked the c code generated by cython, 

```c
typedef npy_float16 __pyx_t_5numpy_float16_t;

static __Pyx_StructField __Pyx_StructFields_nn___pyx_t_5numpy_float16_t[] = {
  {&__Pyx_TypeInfo_nn_npy_uint16, "value", offsetof(__pyx_t_5numpy_float16_t, value)},
  {NULL, NULL, 0}
};
```

I think it's confused why `__pyx_t_5numpy_float16_t` is treated as `unsigned int`, rather than the structure we defined? Do you have any idea? Thanks very much.